### PR TITLE
[Configurable Autosubmit] Implement Validation Filters to determine which validations to run per request type - Part  7

### PR DESCRIPTION
**Pull request by ricardoamador**
Create a Validation Filter class so that we can get the exact filters we want to use depending on the type of pr that is being processed. Since we can configure whether or not to run certain validations this refactor simplifies the code that is needed to determine this.

Note this will not pass check runs until the preceding pull requests are merged up to part 6.

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: auto_submit/lib/validations/validation_filter.dart [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      default:
+        return NoOpValidationFilter(config, repositoryConfiguration);
+    }
+  }
+
+  Set<Validation> getValidations();
+}
+
+/// The [NoOpValidationFilter] allows us to return a set of validations without
+/// exploding in the event something goes wrong.
```
Comment (keyonghan) [inline auto_submit/lib/validations/validation_filter.dart:38]: This is hiding the issue. Shall we through exception explicitly instead if something goes wrong?

Comment (ricardoamador) [reply auto_submit/lib/validations/validation_filter.dart:38]: There really isn't an issue. It is just the default case. I could have just as easily not included it by using if/else. If that is preferable I can change it?

Comment (ricardoamador) [reply auto_submit/lib/validations/validation_filter.dart:38]: I see your point. I will throw if we get here.

File: auto_submit/lib/service/validation_service.dart [hunk 2]
```diff
@@ -39,37 +35,24 @@ class ValidationService {
...
-    final ProcessMethod processMethod = await processPullRequestMethod(messagePullRequest);
+    final RepositoryConfiguration repositoryConfiguration =
+        await config.getRepositoryConfiguration(messagePullRequest.base!.repo!.slug());
+
+    ProcessMethod processMethod;
+    if (!repositoryConfiguration.supportNoReviewReverts) {
+      processMethod = ProcessMethod.processAutosubmit;
+    } else {
+      processMethod = await processPullRequestMethod(messagePullRequest);
+    }
```
[the remark this answers is not in the document]
Comment (ricardoamador) [reply auto_submit/lib/service/validation_service.dart]: Done.

**Pull request merged**